### DISCUSSION TITLE
Allow field or property values to be assigned directly from the constructor parameters

### DISCUSSION THREAD

**Discussion by serhiisolodovnyk** [dotnet/csharplang#7670] [Language Ideas] [answered]
I would like to be able to assign fields and properties in the constructor directly from the parameters. I know that a primary constructor can be a good solution, but sometimes we can't use a primary constructor (for example, when we need to add the `[JsonConstructor]` attribute to the constructor, etc.):

```
public class Car
{
    // current syntax
    public Car(string brand, string model, string color, decimal price)
    {
        Brand = brand;
        Model = model;
        Color = color;
        Price = price;
    }

    // optimal syntax
    public Car(this.Brand, this.Model, this.Color, this.Price)
    {
    }

    public string Brand { get; set; }

    public string Model { get; set; }

    public string Color { get; set; }

    public decimal Price { get; set; }
}

```

This feature was also implemented in Dart with the same syntax:

 ```
class Car {

	     String make;
   	String model;
   	String yearMade;
   	bool hasABS;
   
   	Car(this.make, this.model, this.yearMade, this.hasABS);
}
```

**Comment by yaakov-h** [comment]
Records or primary constructors with #3650 / #7047 seem like they would solve this issue in a better way IMO.

**Comment by DavidArno** [comment, marked as answer]
As of next week, with the release of .NET 8 and C# 12, [you will be able to use primary constructors to express that as](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANACYgDUAPgAIDMABJwCZe6KAgAUnAIwAGXsFqkmDfjN4BbTEwjZlU2fhyYEyrfiJq6vAA4Ii+CAEoAsACgA3q95f+fPbwBCCky8brwA5hAwANy8yJExAL68ALxyQVGe3jwqsgCymtoh4fGxJUmpGlrYGS7ePjnChghFEdGlbeW8BthGNXXZpuaWAAq29i0lcR0p1mMQNQlAA):
```cs
public class Car(string brand, string model, string color, decimal price)
{
    public string Brand { get; set; } = brand;
    public string Model { get; set; } = model;
    public string Color { get; set; } = color;
    public decimal Price { get; set; } = price;
}
```

If you have concerns over this feature, such as the ability to use Json attributes, or you'd prefer to eg be able to express that as:
```cs
public class Car(
    string Brand { get; set; },
    string Model { get; set; },
    string Color { get; set; },
    decimal Price { get; set; })
```
Then head over to #7667 and request it there as Microsoft are actively asking for feedback on this feature.

**Reply by serhiisolodovnyk** [reply]
well, thank you. I discovered that C# 12 will provide the ability to add an attribute to the primary constructor: https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/instance-constructors#primary-constructors
